File: solutions/0941. Valid Mountain Array/0941.java
class Solution {
  public boolean validMountainArray(int[] A) {
    if (A.length < 3)
      return false;

    int l = 0;
    int r = A.length - 1;

    while (l + 1 < A.length && A[l] < A[l + 1])
      ++l;
    while (r > 0 && A[r] < A[r - 1])
      --r;

    return l > 0 && r < A.length - 1 && l == r;
  }
}

File: solutions/0941. Valid Mountain Array/0941.cpp
class Solution {
 public:
  bool validMountainArray(vector<int>& A) {
    if (A.size() < 3)
      return false;

    int l = 0;
    int r = A.size() - 1;

    while (l + 1 < A.size() && A[l] < A[l + 1])
      ++l;
    while (r > 0 && A[r] < A[r - 1])
      --r;

    return l > 0 && r < A.size() - 1 && l == r;
  }
};
